test/unit/teacher_set_test.rb

```diff
@@ -182,13 +186,169 @@ class TeacherSetTest < ActiveSupport::TestCase
+      @elasticsearch_adapter_mock.expect(:update_document_by_id, nil) do
+        raise StandardError, detail_msg
+      end
+    
+      resp = assert_raises(ElasticsearchException) do
+        ElasticSearch.stub :new, @elasticsearch_adapter_mock do
+          @teacher_set.create_or_update_teacherset_document_in_es
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+      @elasticsearch_adapter_mock.verify
+    end
+  end
+
+
+  # Delete teacher-set document in ES
+  describe "Delete teacherset record from elastic search" do
+
+    it 'Delete teacher_set document from Elastic search' do
+      response = {"found" => true, "result" => 'deleted'}
+      @elasticsearch_adapter_mock.expect(:delete_document_by_id, response, [@teacher_set.id])
+      resp = nil
+      ElasticSearch.stub :new, @elasticsearch_adapter_mock do
+        @teacher_set.delete_teacherset_record_from_es
+      end
+      @elasticsearch_adapter_mock.verify
+    end
+
+    it 'raises ElasticsearchException when ElasticSearch raises NotFound' do
+      resp = nil
+      detail_msg = "Teacher set data not found."
+      @elasticsearch_adapter_mock.expect(:delete_document_by_id, nil) do
+        raise Elasticsearch::Transport::Transport::Errors::NotFound, detail_msg
+      end
+
+      resp = assert_raises(ElasticsearchException) do
+        ElasticSearch.stub :new, @elasticsearch_adapter_mock do
+          @teacher_set.delete_teacherset_record_from_es
+        end
+      end
+      assert_equal(TEACHER_SET_NOT_FOUND_IN_ES[:code], resp.code)
+      assert_equal(TEACHER_SET_NOT_FOUND_IN_ES[:msg], resp.message)
+      @elasticsearch_adapter_mock.verify
+    end
+
+    it 'raises ElasticsearchException when ElasticSearch raises StandardError' do
+      resp = nil
+      detail_msg = 'exception occured'
+      @elasticsearch_adapter_mock.expect(:delete_document_by_id, nil) do
+        raise StandardError, detail_msg
+      end
+      resp = assert_raises(ElasticsearchException) do
+        ElasticSearch.stub :new, @elasticsearch_adapter_mock do
+          @teacher_set.delete_teacherset_record_from_es
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+      @elasticsearch_adapter_mock.verify
+    end
+  end
+
+
+  # test teacher set object method.
+  describe "test teacher set object" do
+    it 'test teacher set object' do
+      resp = @teacher_set.teacher_set_info
+      assert_equal(@teacher_set.title, resp[:title])
+      assert_equal(@teacher_set.bnumber, resp[:bnumber])
+      assert_equal(@teacher_set.subjects.count, resp[:subjects].count)
+    end
   end
 
 
+  describe 'update set_type value in teacher_set table' do
+    it 'test update_set_type value method' do
+      resp = nil
+      set_type_val = @teacher_set3.set_type
```

Comment:
I think this would look more meaningful if you changed the `set_type`. Maybe assert what the `@teacher_set3.set_type` equals initially; run the update methods to something else; assert it equals the new value after that.